runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -21,64 +22,65 @@
 @Slf4j
 public class ManagementView extends AbstractView {
 
-    private final long layoutRefreshTimeout = 500;
-
     public ManagementView(@NonNull CorfuRuntime runtime) {
         super(runtime);
     }
 
-    /**
-     * Checks if a given workflow with the given workflow ID is active or not.
-     * If returns True - the workflow is active
-     * If returns False - the workflow is either completed successfully or failed.
-     *
-     * @param server     Node endpoint on which the workflow is running.
-     * @param workflowId Workflow UUID
-     * @return True if active. Else False
-     */
-    public boolean isWorkflowActive(String server, UUID workflowId) {
-        return layoutHelper(l -> {
-            QueryResponse response = runtime.getRouter(server)
-                    .getClient(ManagementClient.class)
-                    .queryRequest(workflowId);
-            log.info("isWorkflowActive: {} on server:{} = {}",
-                workflowId, server, response.isActive());
-            return response.isActive();
-        });
+    void waitForWorkflow(UUID workflow, ManagementClient client) throws NetworkException,
```

Comment:
workflow timeouts should be a parameter here. Probably the number of retries, for that matter too.

Reply:
Number of retries for the timeout to expire, or number of retries of the workflow ? The former is just a function of the query retry and total time to wait.